src/Exceptionless.Core/Repositories/EventRepository.cs

```diff
@@ -127,7 +127,7 @@ public EventRepository(ExceptionlessElasticConfiguration configuration, IValidat
             if (ev == null)
                 return null;
 
-            var retentionDate = Settings.Current.MaximumRetentionDays > 0 ? SystemClock.UtcNow.Date.SubtractDays(Settings.Current.MaximumRetentionDays) : DateTime.MinValue;
+            var retentionDate = AppOptions.Current.MaximumRetentionDays > 0 ? SystemClock.UtcNow.Date.SubtractDays(AppOptions.Current.MaximumRetentionDays) : DateTime.MinValue;
```

Comment:
If these options were added to the Elasticsearch options then you could access them from the configuration that is passed in and not have to use statics for this.

Reply:
To me retention has nothing to do with elastic search

Reply:
It’s max days that we store events in ES as a whole system. It absolutely has to do with ES config. And we are also trying to get rid of static config access as well as try to simplify things.

Reply:
it's also about storage too

Reply:
We are in the repos, we have access to ES config passed into the ctor. It seems pretty straight forward to me that we would want to not be using static config when we can easily use options from the ES config that make perfect sense being in there.